OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/xta/TypeSetEntitySelector.scala

```diff
@@ -30,8 +31,8 @@ object XTASetEntitySelector extends TypeSetEntitySelector {
     override def apply(e: Entity): TypeSetEntity = e match {
         case dm: DefinedMethod        => dm
         case _: VirtualDeclaredMethod => ExternalWorld
-        case f: Field                 => f
-        case _: ExternalField         => ExternalWorld
+        case f: DefinedField          => f.definedField
```

Comment:
As with DefinedMethod, we could probably attach the type sets to the DefinedField directly instead of the Field. That would probably cause further changes in the analyses, though.